sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        """
+        params = {
+            API_FIELD_CLIENT_ID: client_id,
+            API_REQUEST_FIELD_ENRICHMENT_TYPE: str(data_type)
+        }
+        if data_id is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_ID] = data_id
+        if offset is not None:
+            params[API_REQUEST_FIELD_OFFSET] = offset
+        if limit is not None:
+            params[API_REQUEST_FIELD_LIMIT] = limit
+        if subtype is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_SUBTYPE] = str(subtype)
+        if system is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_SYSTEM] = system
+        if company_code is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_COMPANYCODE] = company_code
+
+        response = self.get(DATA_ENDPOINT, params=params,
+                            log_msg_before=f'Getting enrichment data records for client {client_id}')
+        self.logger.info(f'Successfully got {len(response.json()[API_FIELD_VALUE])} enrichment data records')
+        return response.json()[API_FIELD_VALUE]
+
+    def delete_all_enrichment_data(self, data_type=None):
+        """
+        This endpoint is deleting all master data records for the account
+        :param data_type: (Optional) The type of enrichment data that should be deleted. Can be given as
+        :enum:`DoxDataType` or string. Options are 'businessEntity' or 'employee'
+        :return: The API endpoint response as dictionary
+        """
+        delete_url = DATA_ASYNC_ENDPOINT
+
+        params = {API_REQUEST_FIELD_ENRICHMENT_TYPE: str(data_type)} if data_type else None
+        response = self.delete(delete_url, json={API_FIELD_VALUE: []}, params=params,
+                               log_msg_before=f"Start deleting all{f' {data_type}' if data_type else ''} enrichment data records")
+
+        job_id = response.json()[API_FIELD_ID]
+        response = self._poll_for_url(DATA_ID_ENDPOINT.format(id=job_id),
+                                      get_status=lambda r: r[API_FIELD_VALUE][API_FIELD_STATUS],
+                                      log_msg_after=f"Successfully deleted all{f' {data_type}' if data_type else ''} enrichment data records")
+        return response.json()
+
+    def delete_enrichment_data(self, client_id, data_type, payload: list, subtype=None, delete_async=False):
+        """
+        Deletes the enrichment data records with the given IDs in the payload
+        :param client_id: The client ID for which the enrichment data was created
+        :param data_type: The type of enrichment data that should be deleted. Can be given as :enum:`DoxDataType` or
+        string. Options are 'businessEntity' or 'employee'
+        :param payload: A list of dictionaries with the form: ``{'id':'', 'system':'', 'companyCode':''}``
+        :param subtype: (optional) The subtype of the records that should be deleted. Only used for type
```

Comment:
Let's only use either enum or string for this purposes. I would suggest to go with strings for the sake of ease of maintenance, when we add new options we would not have to update the library. When the list of values would be stable we could introduce enums for greater developer comfort.

Reply:
Switched to strings and removed enums